src/components/application_manager/test/hmi_capabilities_test.cc

```diff
@@ -629,45 +1049,314 @@ TEST_F(HMICapabilitiesTest,
+                                                 content_to_save.end());
+  file_system::Write(kHmiCapabilitiesCacheFile, binary_data_to_save);
+
+  EXPECT_TRUE(hmi_capabilities_->SaveCachedCapabilitiesToFile(
+      hmi_interface::tts, sections_to_update, schema_));
+
+  std::string content_after_update;
+  ASSERT_TRUE(
+      file_system::ReadFile(kHmiCapabilitiesCacheFile, content_after_update));
+  EXPECT_TRUE(content_after_update.find(new_language) != std::string::npos);
+}
+
+TEST_F(HMICapabilitiesTest, PrepareJsonValueForSaving_Success) {
+  const std::vector<std::string> sections_to_update{
+      hmi_response::display_capabilities,
+      hmi_response::hmi_zone_capabilities,
+      hmi_response::soft_button_capabilities,
+      strings::audio_pass_thru_capabilities,
+      strings::hmi_capabilities,
+      strings::system_capabilities,
+      hmi_response::languages};
+
+  const std::vector<std::string> interfaces_name{
+      hmi_interface::tts,
+      hmi_interface::vr,
+      hmi_interface::ui,
+      hmi_interface::buttons,
+      hmi_interface::vehicle_info,
+      hmi_interface::rc,
+      hmi_response::speech_capabilities,
+      hmi_response::prerecorded_speech_capabilities,
+      hmi_response::button_capabilities,
+      hmi_response::preset_bank_capabilities};
+
+  smart_objects::SmartObject audio_capabilities_so(
+      smart_objects::SmartType_Array);
+  audio_capabilities_so[0][strings::sampling_rate] =
+      hmi_apis::Common_SamplingRate::RATE_44KHZ;
+  audio_capabilities_so[0][strings::bits_per_sample] =
+      hmi_apis::Common_BitsPerSample::RATE_8_BIT;
+  audio_capabilities_so[0][strings::audio_type] =
+      hmi_apis::Common_AudioType::PCM;
+
+  const std::string content_to_save = "{\"field\" : \"value\" }";
+  const std::vector<uint8_t> binary_data_to_save(content_to_save.begin(),
+                                                 content_to_save.end());
+
+  hmi_capabilities_->set_audio_pass_thru_capabilities(audio_capabilities_so);
+
+  CreateFile(kHmiCapabilitiesCacheFile);
+  file_system::Write(kHmiCapabilitiesCacheFile, binary_data_to_save);
+
+  for (size_t i = 0; i < interfaces_name.size(); ++i) {
+    EXPECT_TRUE(hmi_capabilities_->SaveCachedCapabilitiesToFile(
+        interfaces_name[i], sections_to_update, schema_));
+  }
+
+  std::string content_after_update;
+  EXPECT_TRUE(
+      file_system::ReadFile(kHmiCapabilitiesCacheFile, content_after_update));
+
+  Json::Value root_node;
+  utils::JsonReader reader;
+  ASSERT_TRUE(reader.parse(content_after_update, &root_node));
+
+  for (size_t i = 0; i < interfaces_name.size(); ++i) {
+    EXPECT_TRUE(static_cast<bool>(root_node[interfaces_name[i]]));
+  }
+  EXPECT_TRUE(
+      root_node[hmi_interface::ui][strings::audio_pass_thru_capabilities]
+          .isArray());
+}
+
+TEST_F(HMICapabilitiesTest,
+       OnCapabilityInitialized_RespondToAllPendingRAIRequestsIfTheyHold) {
+  // Contains only UI capabilities
```

Comment:
if an entire interface is missing from the capabilities file, should we send get capabilities requests for that interface? I asked this earlier with the interfaceNode.isNull() json checks in LoadCapabilitiesFromFile

Reply:
@iCollin Changed in the cb6e7d550aa0b17533d3d6fa655575e55cf5b7a2